src/Symfony/Component/String/AbstractUnicodeString.php

```diff
@@ -587,4 +623,21 @@ private function wcswidth(string $string): int
 
         return $width;
     }
+
+    private function getLocaleTransliterator(string $locale, string $id): ?\Transliterator
+    {
+        $rule = $locale.'-'.$id;
+        if (null !== $transliterator = self::$transliterators[$rule] ??= \Transliterator::create($rule)) {
```

Comment:
we need to cache when create() returns
this means we should use array_key_exists instead (same at the end of the method)

Reply:
You're right, fixes with latest commit:
```php
$string->localeTitle("nl");
$string->localeTitle("nl_BE");
$string->localeTitle("nl_BE");
```
NL: creates and caches NL transliterator
NL_BE: Unable to create NL_BE, uses NL from cache, caches as NL_BE
NL_BE: Uses NL_BE from cache